### ISSUE TITLE
Create Transformer model

### ISSUE THREAD

**Issue by Kairixir** [Kairixir/ml-tartu-electricity-prediciton#13]

**Comment by Kairixir** [comment]
I've found a parallelisation issue. So far I found this [issue](https://github.com/pytorch/pytorch/issues/68407) where a possible solution offers changing the Positional encoding of Transformer. 

I think I could get into the guts of [Darts](https://unit8co.github.io/darts) and change the positional encoding by creating a whole new [custom encoder and decoder](https://unit8co.github.io/darts/generated_api/darts.models.forecasting.transformer_model.html). Would be nice to have the Darts implementation though. Maybe I can find it in their GitHub.

Another approach can be to find different way of parallelisation. I found [this](https://towardsdatascience.com/parallel-inference-of-huggingface-transformers-on-cpus-4487c28abe23) article which shows how to parallelise [huggingface implementation of transformer](https://huggingface.co/docs/transformers/index) using [Ray](https://github.com/ray-project/ray) library

P.S. Playing with the [Trainer](https://pytorch-lightning.readthedocs.io/en/latest/common/trainer.html?highlight=CPU%20) in Darts Transformer class definition did not help. I guess it was tested only on GPUs

**Comment by Kairixir** [comment]
Can be worth to instantiate [learning rate decay](https://pytorch.org/docs/stable/generated/torch.optim.lr_scheduler.StepLR.html) to improve loss at later part of learning [(logic behing it)](https://stackoverflow.com/questions/39517431/should-we-do-learning-rate-decay-for-adam-optimizer)

**Comment by Kairixir** [comment]
Could be worth to implement [early stopping](https://pytorch-lightning.readthedocs.io/en/stable/common/early_stopping.html) so the model is not overfit (also decreases number of epochs, yay!) 
Details for this can be found at [Darts Transformer model page](https://unit8co.github.io/darts/generated_api/darts.models.forecasting.transformer_model.html) at argument "pl_trainer_kwargs"

**Comment by Kairixir** [comment]
Guide for [optimisation tuning in PyTorch](https://pytorch.org/tutorials/recipes/recipes/tuning_guide.html)
